pkg/controller/replicaset/replica_set.go

```diff
@@ -122,6 +126,23 @@ func NewReplicaSetController(kubeClient clientset.Interface, resyncPeriod contro
 		framework.ResourceEventHandlerFuncs{
 			AddFunc: rsc.enqueueReplicaSet,
 			UpdateFunc: func(old, cur interface{}) {
+				oldRS := old.(*extensions.ReplicaSet)
+				curRS := cur.(*extensions.ReplicaSet)
+
+				// We should invalid the whole lookup cache if a RS's selector has been updated.
+				//
+				// Imagine that you have two RS:
+				// * old RS1
+				// * new RS2 You also have a pod that is attached to RS2 (because it doesn't match RS1 selector).
```

Comment:
nit: can you make those lines shorter (i.e. split them differently) so that it is easier to read it? Thanks!